### ISSUE TITLE
Release 8.5.0 - Failed building wheel for cffi

### ISSUE THREAD

**Issue by stevejr** [squidfunk/mkdocs-material#4366]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Our GH Action started to fail when using version 8.5.0 of mkdocs-material. The error appears to be with the package cffi, specifically this is what we are seeing:

```
running build_ext
            building '_cffi_backend' extension
            creating build/temp.linux-x86_64-3.8
            creating build/temp.linux-x86_64-3.8/c
            gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -DTHREAD_STACK_SIZE=0x100000 -fPIC -DFFI_BUILDING=1 -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/usr/include/ffi -I/usr/include/libffi -I/usr/local/include/python3.8 -c c/_cffi_backend.c -o build/temp.linux-x86_64-3.8/c/_cffi_backend.o
            c/_cffi_backend.c:15:10: fatal error: ffi.h: No such file or directory
               15 | #include <ffi.h>
                  |          ^~~~~~~
            compilation terminated.
            error: command 'gcc' failed with exit status 1
            [end of output]
```

### Expected behaviour

Release 8.5.0 should work

### Actual behaviour

Release 8.5.0 is failing when used

### Steps to reproduce

Build from a Dockerfile:

```
ARG PYTHON_VERSION=3.8.1-alpine3.11

FROM python:${PYTHON_VERSION} as builder

ENV PYTHONUNBUFFERED 1

# Set build directory
WORKDIR /wheels

# Copy files necessary
COPY ./requirements.txt .

# Perform build and cleanup artifacts
RUN \
  apk add --no-cache \
    git \
    git-fast-import \
  && apk add --no-cache --virtual .build gcc musl-dev \
  && python -m pip install --upgrade pip \
  && pip install -r requirements.txt \
  && apk del .build gcc musl-dev \
  && rm -rf /usr/local/lib/python3.8/site-packages/mkdocs/themes/*/* \
  && rm -rf /tmp/*

# Set final MkDocs working directory
WORKDIR /docs

# Expose MkDocs development server port
EXPOSE 8000

# Start development server by default
ENTRYPOINT ["mkdocs"]
CMD ["serve", "--dev-addr=0.0.0.0:8000"]
```

requirements.txt is:

```
# Direct dependencies
mkdocs>=1.2
Pygments>=2.4
markdown>=3.2
pymdown-extensions>=7.0
mkdocs-material-extensions>=1.0
mkdocs-git-revision-date-localized-plugin>=0.7.3
mkdocs-material>=7.1.8 
```

### Package versions

- Python: `3.9`
- MkDocs: `mkdocs-1.3.1`
- Material: `8.5.0`

### Configuration

```yaml
https://docs.k8slens.dev
```

### System information

- Operating system: N/A
- Browser: N/A

**Comment by squidfunk** [comment]
Thanks for reporting. This is due to the new requirement for `cairosvg` to allow for social cards. However, we should make this dependency a peer dependency to be installed by the user, so Material for MkDocs builds without native extensions.

**Comment by stevejr** [comment]
No problem. Is there something I can do to our build process to work around the issue?

**Comment by squidfunk** [comment]
Yes, downgrade to 8.4.4:

```
pip install "mkdocs-material<8.5"
```

**Comment by squidfunk** [comment]
Fixed in 5afacad66c0e983535cd51cc5f75a3897162d0d8. I've removed `pillow` and `cairosvg` from the requirements of Material for MkDocs. If the social plugin is added to `plugins`, it will check if those libraries are installed, and terminate with an error if they aren't:

```
ERROR    -  Required dependencies of "social" plugin not found. Install with: pip install pillow cairosvg
```

It is now the obligation of the user to install the required libraries. This is now also mentioned in the documentation:
https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#dependencies
